lib/search-bar.jsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import appState from './flux/app-state';
+import { tracks } from './analytics';
+import NewNoteIcon from './icons/new-note';
+import SearchField from './search-field';
+import TagsIcon from './icons/tags';
+
+export const SearchBar = ( {
+	onNewNote,
+	onToggleNavigation,
+	showTrash,
+} ) =>
+	<div className="search-bar theme-color-border">
+		<button className="button button-borderless" onClick={ onToggleNavigation } title="Tags">
+			<TagsIcon />
+		</button>
+		<SearchField />
+		<button
+			className="button button-borderless"
+			disabled={ showTrash }
+			onClick={ onNewNote }
+			title="New Note"
+		>
+			<NewNoteIcon />
+		</button>
+	</div>;
+
+const {
+	newNote,
+	toggleNavigation,
+} = appState.actionCreators;
+const { recordEvent } = tracks;
```

Comment:
Curiously enough, I got to think about this a couple of hours ago!
My reasoning would be to keep the Redux stuff close to each other, but it feels a bit weird indeed.
I'm gonna move them up no problems.